deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -326,11 +326,11 @@ def search(
         )
 
         exec_option = exec_option or self.exec_option
-        embedding_function = embedding_function or self.embedding_function
 
         utils.parse_search_args(
             embedding_data=embedding_data,
             embedding_function=embedding_function,
+            initial_embedding_function=self.embedding_function,
```

Comment:
is this used to change the embedding function?

Reply:
the idea is that embedding function can be specified from the very beginning, which is called `initial_embedding_function `